### PR TITLE
Refactor and Update for Spark 3, Add TLS and auth tokens

### PR DESCRIPTION
**Pull request by kdbrooks**
PR to update this to Spark 3 and the Data Source V2 API.  It also adds support for bearer tokens and TLS.

This was tested using Databricks 10.4 LTS and Arrow Datafusion as the SQL engine on the flight server.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-dist: xenial
```
Comment (rymurr) [inline .travis.yml:1]: nit: maybe we want to change to gh actions? I am not sure if travis still works?

File: src/main/java/org/apache/arrow/flight/spark/FlightClientOptions.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2019 Ryan Murray
```
Comment (rymurr) [inline src/main/java/org/apache/arrow/flight/spark/FlightClientOptions.java]: want to update the license header to include you or just `authors`

**Review by rymurr — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (kdbrooks) [reply .travis.yml:1]: Agreed, I want to switch to gh actions and also publish the JAR to maven central.  I'll plan on doing this in another PR.

File: src/main/java/org/apache/arrow/flight/spark/FlightClientOptions.java (hunk 2 above)
Comment (mattphelps-8451) [reply src/main/java/org/apache/arrow/flight/spark/FlightClientOptions.java]: @rymurr Kyle included you, himself, and Doron in authors https://github.com/rymurr/flight-spark-source/pull/8/commits/55f804a933c1beeb7b8731756f047b73f5dd95b0

**Pull request merged**